mantidproject/mantid
# Remove shortcuts when uninstalling

**Description of work.**
A workaround for a NSIS/Windows 10 bug where uninstalling via the "Apps & features" list asks for an admin login to run the uninstaller. This would cause the uninstaller to run under the admin account, so it would fail to delete the shortcuts (which are installed the local user's desktop and start menu). Note that the same issue is not present in the old legacy version of the programs list found via the control panel. The uninstaller will also function correctly if run directly.

The workaround will restart the uninstaller .exe as the current Windows user. A couple of NSIS plugins are required, so the third party repo has been updated, which will also have to be merged:
https://github.com/mantidproject/thirdparty-msvc2015/pull/63

**To test:**
Perform these steps in a non-admin Windows account
- Download and run the installer from the Windows build:
https://builds.mantidproject.org/job/pull_requests-windows/44625/artifact/build/mantidunstable-6.1.20210624.1500-win64.exe
(you may need to rebuild with BUILD_PACKAGE turned on and wait for a new installer to be built if this one has disappeared, or contact me and I'll point you to a shared drive with it on)
- Once installed, navigate to the Windows 10 "Apps & features" list.
- Click on "Mantid Unstable" and "Uninstall".
- When prompted, enter an admin username and password.
- Verify that the desktop and start menu shortcuts have been removed.


Fixes #31671 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
